static/widgets/load-save.ts

```diff
@@ -91,17 +99,20 @@ export class LoadSave {
         this.modal?.modal('hide');
     }
 
-    private static populate(root: JQuery, list: {name: string, load: () => void}[]) {
+    private static populate(root: JQuery, list: {name: string, load: () => void, delete?: () => void }[]) {
         root.find('li:not(.template)').remove();
         const template = root.find('.template');
         for (const elem of list) {
-            template
-                .clone()
-                .removeClass('template')
+            const clone = template.clone();
+            clone.removeClass('template')
                 .appendTo(root)
                 .find('a')
                 .text(elem.name)
                 .on('click', elem.load);
+            const deleteButton = clone.find('button');
+            if (deleteButton && elem.delete !== undefined) {
+                deleteButton.on('click', () => elem.delete?.());
```

Comment:
I feel like TS ought to know that elem is not undefined at this point, but it was complaning, hence the `?.()`

Reply:
I think TS is right here, because the if-check cannot guarantee that elem.delete is not modified after the if-check because it's in a callback function